workerTypes/index.d.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Context } from '@azure/functions';
+
+/**
+ * This module is shipped as a built-in part of the Azure Functions Node.js worker and is available at runtime
+ */
+declare module '@azure/functions-worker' {
+    /**
+     * Register a hook to interact with the lifecycle of Azure Functions.
+     * Hooks are executed in the order they were registered and will block execution if they throw an error
+     */
+    export function registerHook(hookName: 'preInvocation', callback: PreInvocationCallback): Disposable;
+    export function registerHook(hookName: 'postInvocation', callback: PostInvocationCallback): Disposable;
+    export function registerHook(hookName: string, callback: HookCallback): Disposable;
+
+    export type HookCallback = (context: HookContext) => void | Promise<void>;
+    export type PreInvocationCallback = (context: PreInvocationContext) => void | Promise<void>;
+    export type PostInvocationCallback = (context: PostInvocationContext) => void | Promise<void>;
+
+    export type HookData = { [key: string]: any };
+
+    /**
+     * Base interface for all hook context objects
+     */
+    export interface HookContext {
+        /**
+         * The recommended place to share data between hooks
+         */
+        hookData: HookData;
```

Comment:
we do not want `invocationContext: Context;` in the `HookContext` despite children of the interface have it, right?

Reply:
Correct, because eventually we will likely have hooks unrelated to invocation (like worker startup/teardown tracked by https://github.com/Azure/azure-functions-nodejs-worker/issues/539)